app/src/main/kotlin/com/wire/android/ui/newauthentication/welcome/NewWelcomeScreen.kt

```diff
@@ -0,0 +1,115 @@
+/*
+ * Wire
+ * Copyright (C) 2025 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ */
+
+@file:Suppress("TooManyFunctions")
+
+package com.wire.android.ui.newauthentication.welcome
+
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.ui.Modifier
+import androidx.hilt.navigation.compose.hiltViewModel
+import com.ramcosta.composedestinations.annotation.RootNavGraph
+import com.wire.android.BuildConfig
+import com.wire.android.navigation.BackStackMode
+import com.wire.android.navigation.NavigationCommand
+import com.wire.android.navigation.Navigator
+import com.wire.android.navigation.WireDestination
+import com.wire.android.navigation.style.AuthPopUpNavigationAnimation
+import com.wire.android.ui.authentication.login.LoginNavArgs
+import com.wire.android.ui.authentication.login.WireAuthBackgroundLayout
+import com.wire.android.ui.authentication.welcome.WelcomeScreenState
+import com.wire.android.ui.authentication.welcome.WelcomeViewModel
+import com.wire.android.ui.common.dialogs.MaxAccountsReachedDialog
+import com.wire.android.ui.common.dialogs.MaxAccountsReachedDialogState
+import com.wire.android.ui.common.preview.EdgeToEdgePreview
+import com.wire.android.ui.common.visbility.rememberVisibilityState
+import com.wire.android.ui.destinations.NewLoginScreenDestination
+import com.wire.android.ui.destinations.NewWelcomeScreenDestination
+import com.wire.android.ui.destinations.WelcomeScreenDestination
+import com.wire.android.ui.theme.WireTheme
+import com.wire.android.util.ui.PreviewMultipleThemes
+import com.wire.kalium.logic.configuration.server.ServerConfig
+import kotlinx.coroutines.delay
+
+@RootNavGraph(start = true)
+@WireDestination(
+    navArgsDelegate = LoginNavArgs::class
+)
+@Composable
+// this is a temporary solution because annotation argument "start" must be a compile-time constant
```

Comment:
Good trick ! 

Question: can we maintain this in the future ? Mainly because we will need to resolve this not by compile time, but with API version based. If the backend is v8+ then new login flow. If not, the old/current flow.

Reply:
Oh, so in short: yes, we can do that!

In that case we would need to create a small dedicated `ViewModel` that uses `AuthServerConfigProvider` and inject it into composable to determine the login flow version to use and this way we create a screen destination that works as a login flow "chooser", but we would need to create such composable "chooser" for the "welcome" and "login" screen (sometimes we directly open "login" screen, e.g. when migration requires the user to log in).